controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -218,16 +214,6 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		// Configuration phase.
 		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimConfiguring
 
-		// Check if finalizers are present, if not, add them.
-		if !contains(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer) {
-			storageClassClaimRef := klog.KRef(r.storageClassClaim.Name, r.storageClassClaim.Namespace)
-			r.log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", storageClassClaimRef)
-			r.storageClassClaim.SetFinalizers(append(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer))
-			if err := r.update(r.storageClassClaim); err != nil {
-				return reconcile.Result{}, fmt.Errorf("failed to update StorageClassClaim [%v] with finalizer: %s", storageClassClaimRef, err)
-			}
-		}
```

Comment:
Why are we removing finalizers from consumer phases? in the consumer, the claim finalizer is needed in order to ensure a revoke call to the provider (which is not. k8s construct and cannot be ensured by k8s delete mechinizem without a finalizer). Please revert this change.